### PR TITLE
feat: coupon loading more dynamic

### PR DESCRIPTION
**Pull request by joelhooks**
dynamic

### CODE REVIEW

File: packages/commerce-server/src/props-for-commerce.ts [hunk 1]
```diff
@@ -24,9 +24,11 @@ export async function propsForCommerce({
     ? await getPurchasesForUser(token.sub as string)
     : false
 
+  console.log({products})
```
Comment (Creeland) [inline packages/commerce-server/src/props-for-commerce.ts:27]: Do we want to keep this console log?

File: packages/commerce-server/src/get-active-merchant-coupon.ts [hunk 2]
```diff
@@ -14,7 +14,9 @@ export async function getActiveMerchantCoupon({
 
   let activeMerchantCoupon = null
 
-  const defaultCoupons = await getDefaultCoupon(productId)
+  const defaultCoupons = productId
+    ? await getDefaultCoupon([productId])
```
Comment (Creeland) [inline packages/commerce-server/src/get-active-merchant-coupon.ts:18]: Is getDefaultCoupon capable of accepting an array as an argument? [async getDefaultCoupon(productId?: string) {](https://github.com/skillrecordings/products/blob/d4c420f10094235a96a923ec2d3afc547276acd5/packages/database/src/prisma-api.ts#L644-L667)

**Review by Creeland — COMMENTED** [review]
So, if I'm understanding the changes correctly. This now checks if one of several coupons

File: packages/commerce-server/src/get-active-merchant-coupon.ts (hunk 2 above)
Comment (joelhooks) [reply packages/commerce-server/src/get-active-merchant-coupon.ts:18]: as part of this PR is is now

File: packages/commerce-server/src/props-for-commerce.ts (hunk 1 above)
Comment (joelhooks) [inline packages/commerce-server/src/props-for-commerce.ts:27]:
```suggestion
```

**Review by vojtaholik — APPROVED** [review]

**Pull request merged**